hoot-core/src/main/cpp/hoot/core/io/OsmChangeWriter.cpp

```diff
@@ -40,56 +44,50 @@ void OsmChangeWriter::write(QIODevice &d, ChangeSetProviderPtr cs)
   writer.writeAttribute("version", "0.6");
   writer.writeAttribute("generator", "hootenanny");
 
-  Change::ChangeType last = Change::Unknown;
-
+  //Make the changeset have only one set of entries per change type.  Also for readability,
+  //sort the set of changes within each change type by element type.
+  QMap<Change::ChangeType, QMap<ElementType::Type, QVector<ConstElementPtr> > > elementsByChangesetTypeAndElementType;
```

Comment:
That's a good point...hadn't thought of that.  The problem I was trying to address is that before the change the OsmChangeWriter was writing multiple instances of each change type (create/update/delete) to the output xml changeset document.  I don't believe the rails port will accept that...but maybe I'm wrong....need to look more closely at the rails port code.